FrontEnd/src/UserAuthentication/FirestoreDB.js

```diff
@@ -1,16 +1,85 @@
-import {database} from './FirebaseConfig';
-import {doc, setDoc} from 'firebase/firestore';
+import { database, auth } from "./FirebaseConfig";
+import {
+  query,
+  where,
+  collection,
+  doc,
+  getDocs,
+  setDoc,
+  getDoc,
+} from "firebase/firestore";
+import { onAuthStateChanged } from 'firebase/auth';
+import {useState, useEffect} from 'react';
 
-async function addUser(userCredential, firstName, lastName, userName, signUpEmail){
-    const uid = userCredential.user.uid;
-    const userDocRef = doc(database, "users", uid);
-    await setDoc(userDocRef, {
-        uid: uid,
-        FirstName: firstName,
-        LastName: lastName,
-        UserName: userName,
-        Email: signUpEmail,
+async function addUser(userId, firstName, lastName, userName, signUpEmail) {
+  const userDocRef = doc(database, "users", userId);
+  return await setDoc(userDocRef, {
+    userID: userId,
+    FirstName: firstName,
+    LastName: lastName,
+    UserName: userName,
+    Email: signUpEmail,
+  });
+}
+
+async function isDatabase() {
+  const usersCollection = query(collection(database, "users"));
+  const querySnapshot = await getDocs(usersCollection);
+  return !querySnapshot.empty;
+}
+
+async function isUsernameUnique(username) {
+  const databaseExists = await isDatabase("users");
+  console.log(databaseExists);
```

Comment:
Don't commit `console.log` statements. It's OK to use it during development though.